//
// Aspia Project
// Copyright (C) 2018 Dmitry Chapyshev <dmitry@aspia.ru>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/>.
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "common.proto";

package proto.host;

message ConnectEvent
{
    string uuid              = 1;
    string remote_address    = 2;
    string username          = 3;
    SessionType session_type = 4;
}

message DisconnectEvent
{
    string uuid = 1;
}

message KillSession
{
    string uuid = 1;
}

message UiToService
{
    KillSession kill_session = 1;
}

message ServiceToUi
{
    ConnectEvent connect_event       = 1;
    DisconnectEvent disconnect_event = 2;
}
